service/src/main/java/com/epam/esm/dto/mapper/CertificateMapper.java

```diff
@@ -0,0 +1,49 @@
+package com.epam.esm.dto.mapper;
+
+import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.entity.Certificate;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.repository.repositoryinterfaces.TagRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class CertificateMapper {
+
+    private CertificateMapper() {
+    }
+
+    public List<CertificateDto> changeListOfCertificatesToDto(List<Certificate> certificates, TagRepository tagRepository) {
```

Comment:
Your mapper is dedicated to converting one dataclass to another. It shouldn't use repositort or any other domain classes at all. You can pass list of certificates and Map<String, List<Tag>> with certificateId as key

Reply:
fixed